internal/pkg/handler/upgrade.go

```diff
@@ -147,6 +147,18 @@ func getContainerToUpdate(volumes []v1.Volume, containers []v1.Container, envarP
 		}
 	}
 
+	// Get the container with referenced secret or configmap
+	for i := range containers {
```

Comment:
I think we need to remove the duplication from these 2 loops. The same has been done above, almost the same actually. Please try to common out the logic

Reply:
Updated, now it is using a single container loop